opendatahub-io/ai-edge
# RHOAIENG-6869: Support running tests based on values given to the e2e tests

## Description
This PR aims to make the go e2e test more configurable and more usable for cases in which you want to test but don't have all the requirements for the different types of tests.

Using a `config.json` the values are passed to the e2e tests. Based on the this config the e2e tests may or may not run certain tests, this is set from the `enabled` field in each sub object in the config.
```json
{
  "namespace": "",
  "image_registry_username": "",
  "image_registry_password": "",
  "target_image_tags": [],

  "git_fetch": {
    "enabled": false,
    "container_file_repo": "",
    "container_file_revision": "",
    "container_relative_path": "",
    "model_repo": "",
    "model_relative_path": "",
    "model_revision": "",
    "model_dir": "",
    "self_signed_cert": ""
  },

  "s3_fetch": {
    "enabled": false,
    "aws_secret": "",
    "aws_access": "",
    "region": "",
    "endpoint": "",
    "bucket_name": "",
    "self_signed_cert": ""
  },

  "gitops": {
    "enabled": false,
    "token": "",
    "username": "",
    "repo": "",
    "api_server": "",
    "branch": ""
  }
}

```

For example, if you are only trying to run the main MLOps pipeline it is not needed to give the values for gitops. By setting the `enabled` field you can set the gitops pipeline not to run.

The `config.json` is structured to have all non-specific values at the root (Namespace etc.) and then all optional values for tests in their own sub object, here labelled `s3_fetch` and `git_fetch` as they are the two options for fetching. 

## How Has This Been Tested?
- Update the config.json file in the e2e test folder with the values required, the same values used before should work as normal
- Run `make go-test`
- All tests should pass
- Remove the values each fetch or the gitops fields and see how the e2e tests change their behaviour
- All tests should pass again

## Merge criteria:
